value.ml

```diff
@@ -764,3 +764,16 @@ and value_of_xmlitem =
     | Text s -> `Variant ("Text", box_string s)
     | Attr (name, value) -> `Variant ("Attr", `Record [("1", box_string name); ("2", box_string value)])
     | Node (name, children) -> `Variant ("Node", `Record [("1", box_string name); ("2", value_of_xml children)])
+
+let rec xml_of_variants vs = match vs with
+  | (`List variant_items) -> List.map xmlitem_of_variant variant_items
+  | _ -> failwith "Cannot construct xml from variants"
+and xmlitem_of_variant =
```

Comment:
exposing the xmlitem version of this as a Links-callable function would make it possible to avoid some oercions to/from singleton lists in the library calls...